Ramda/Extensions/empty.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  empty.swift
+//  Ramda
+//
+//  Created by Christian Leovido on 27/09/2019.
+//
+
+import Foundation
+
+public protocol EmptyDefinable {
+
+    associatedtype CollectionType: Collection
+    associatedtype SubCollectionType: Collection
+
+    mutating func emptyCollection() -> CollectionType
+}
+
+extension R {
+
+    /**
+
+     Returns the empty value of its argument's type.
+
+     - parameter value: must conform to the EmptyDefinable protocol
+
+     - returns: an empty collection type (Array, Dictionary, Set, String)
+
+     */
+
+    public class func empty<T: EmptyDefinable>(value: T) -> T? {
+        var incomingEmptyDefinable = value
+
+        return incomingEmptyDefinable.emptyCollection() as? T
+    }
+
+}
+
+extension Array: EmptyDefinable {
+
+    public typealias CollectionType = [Any]
```

Comment:
Also would there be a way to keep the array element generic rather so instead of `[Any]` rather make it equal `[Element]`?